kernel/src/process.rs

```diff
@@ -2075,6 +2075,28 @@ impl<C: 'static + Chip> Process<'_, C> {
         self.allow_high_water_mark
             .set(self.original_allow_high_water_mark);
 
+        // Reset the MPU configuration based on the original app_break, which is
+        // likely different than what the app break was when the app crashed.
+        let mpu_config_res = self.mpu_config.map_or(Err(()), |mut config| {
+            self.chip.mpu().update_app_memory_region(
+                self.app_break.get(),
+                self.kernel_memory_break.get(),
```

Comment:
```suggestion
                self.original_app_break,
                self.original_kernel_memory_break,
```
(I realize this is basically equivalent, because `app_break` is set to `original_app_break` above, but this way removes an extra `get()` call and protects this mistake from being recreated if the code in this function is shuffled such that the MPU reconfig happens before changing the non-original values)